packages/push-signal/src/repositories/interop.repository.ts

```diff
@@ -2,35 +2,30 @@ import { genericError, DB, TableName } from "pagopa-signalhub-commons";
 import { config } from "../config/env.js";
 
 export interface IInteropRepository {
-  findBy: (
+  findByEserviceIdAndProducerId: (
     eserviceId: string,
-    purposeId: string,
-    purposeState: string,
-    eserviceState: string
-  ) => Promise<{
-    eservice?: { id: string; state: string; producerId: string };
-  } | null>;
+    producerId: string,
+    state: string
+  ) => Promise<string | null>;
 }
 
 export const interopRepository = (db: DB): IInteropRepository => {
   const eserviceTable: TableName = `${config.interopSchema}.eservice`;
-  const purposeTable: TableName = `${config.interopSchema}.purpose`;
   return {
-    async findBy(
+    async findByEserviceIdAndProducerId(
```

Comment:
I prefer "findBy" and leave the parameter complete the "sentence". But it's only my preference :)
Never mind.